### PR TITLE
Document event resource with swagger

### PR DESCRIPTION
**Pull request by gionn**

### CODE REVIEW

File: cmw-dto/src/main/java/com/liberologico/cloudesire/cmw/model/dto/BaseEntityDTO.java [hunk 1]
```diff
@@ -13,15 +13,14 @@
     @ApiModelProperty( readOnly = true )
     private Integer id;
 
-    @ApiModelProperty( readOnly = true )
+    @ApiModelProperty( readOnly = true, example = "Subscription/123")
```
Comment (malteo) [inline cmw-dto/src/main/java/com/liberologico/cloudesire/cmw/model/dto/BaseEntityDTO.java]: `subscription` lowercase

(also watch the formatting)

File: cmw-dto/src/main/java/com/liberologico/cloudesire/cmw/model/dto/EventDTO.java [hunk 2]
```diff
@@ -3,23 +3,37 @@
...
 /**
- * This is the object exposed by the API, not the notification sent
+ * This is the object exposed by the API, not the notification sent via HTTP
  *
  * @see EventNotificationDTO
  */
+@ApiModel( "An event dispatched by the platform to a recipient" )
 public class EventDTO extends BaseEntityDTO
 {
+    @ApiModelProperty( value = "The name of the entity", example = "Subscription" )
```
Comment (malteo) [inline cmw-dto/src/main/java/com/liberologico/cloudesire/cmw/model/dto/EventDTO.java]: I'm not sure about this if it's lowercase too or not

File: cmw-dto/src/main/java/com/liberologico/cloudesire/cmw/model/dto/EventDTO.java [hunk 3]
```diff
@@ -3,23 +3,37 @@
...
     private String entityName;
-    private Integer entityId; // All events are entity specific
+
+    @ApiModelProperty( value = "The Id of the entity", example = "123" )
+    private Integer entityId;
+
+    @ApiModelProperty( "If the platform has successfully dispatched this event" )
```
Comment (malteo) [inline cmw-dto/src/main/java/com/liberologico/cloudesire/cmw/model/dto/EventDTO.java]: `readOnly`?

File: cmw-dto/src/main/java/com/liberologico/cloudesire/cmw/model/dto/EventDTO.java [hunk 4]
```diff
@@ -3,23 +3,37 @@
...
     private Boolean notified = false;
+
+    @ApiModelProperty( value = "When the event has been generated", readOnly = true )
     private Date date = new Date();
```
Comment (malteo) [inline cmw-dto/src/main/java/com/liberologico/cloudesire/cmw/model/dto/EventDTO.java]: won't this initialization mess with the converter?

File: cmw-dto/src/main/java/com/liberologico/cloudesire/cmw/model/dto/EventNotificationDTO.java [hunk 5]
```diff
@@ -12,13 +14,24 @@
  *
  * @see EventDTO
  */
+@ApiModel( "Payload sent to the HTTP endpoint" )
 public class EventNotificationDTO extends BaseEntityDTO
 {
+    @ApiModelProperty( value = "The Id of the Event", example = "123")
```
Comment (malteo) [inline cmw-dto/src/main/java/com/liberologico/cloudesire/cmw/model/dto/EventNotificationDTO.java]: we used `ID` all uppercase elsewhere

the spaceeee!

File: cmw-dto/src/main/java/com/liberologico/cloudesire/cmw/model/dto/EventNotificationDTO.java [hunk 6]
```diff
@@ -12,13 +14,24 @@
...
     private Integer eventId;
+
+    @ApiModelProperty( value = "The name of the Event", example = "Subscription" )
```
Comment (malteo) [inline cmw-dto/src/main/java/com/liberologico/cloudesire/cmw/model/dto/EventNotificationDTO.java]: same as above

File: cmw-dto/src/main/java/com/liberologico/cloudesire/cmw/model/dto/EventNotificationDTO.java [hunk 7]
```diff
@@ -12,13 +14,24 @@
...
     private String entity;
+
+    @ApiModelProperty( value = "The date of Event creation" )
     private Date date;
+
     private CmwEventType type;
+
+    @ApiModelProperty( value = "Concatenation of entity name and id", example = "Subscription/123" )
```
Comment (malteo) [inline cmw-dto/src/main/java/com/liberologico/cloudesire/cmw/model/dto/EventNotificationDTO.java]: same as above (not sure)

File: cmw-dto/src/main/java/com/liberologico/cloudesire/cmw/model/enums/EventRecipientType.java [hunk 8]
```diff
@@ -1,5 +1,8 @@
 package com.liberologico.cloudesire.cmw.model.enums;
 
+import io.swagger.annotations.ApiModel;
+
+@ApiModel( "If the recipient of the event is a company or a platform endpoint" )
```
Comment (malteo) [inline cmw-dto/src/main/java/com/liberologico/cloudesire/cmw/model/enums/EventRecipientType.java]: "Destination of the event"

File: cmw-dto/src/main/java/com/liberologico/cloudesire/cmw/model/enums/EventRecipientType.java [hunk 9]
```diff
@@ -1,5 +1,8 @@
...
 public enum EventRecipientType
 {
     COMPANY, PLATFORM
```
Comment (malteo) [inline cmw-dto/src/main/java/com/liberologico/cloudesire/cmw/model/enums/EventRecipientType.java]: COMPANY - "The event recipient is an endpoint defined by the vendor"
PLATFORM - "The event recipient is a platform endpoint"

File: cmw-dto/src/main/java/com/liberologico/cloudesire/cmw/model/dto/EventDTO.java (hunk 2 above)
Comment (gionn) [reply cmw-dto/src/main/java/com/liberologico/cloudesire/cmw/model/dto/EventDTO.java]: https://docs.cloudesire.com/docs/syndication.html#anatomy-of-an-event-request

**Pull request merged**